AK/SIMDExtras.h

```diff
@@ -253,6 +253,40 @@ ALWAYS_INLINE static T elementwise_byte_reverse_impl(T a, IndexSequence<Idx...>)
     }
 }
 
+template<SIMDVector T, size_t... Idx>
+ALWAYS_INLINE static ElementOf<T> reduce_or_impl(T const& a, IndexSequence<Idx...> const&)
+{
+    static_assert(is_power_of_two(vector_length<T>));
```

Comment:
Technically not a requirement, just  a limitation of the generic impl,
Clang officially uses an even-odd pattern for their builtin

Reply:
Yes, you are correct. But I in practice there is always power of two. And I was not feeling like implementing general case. Might add /*todo*/.